wrangler-service/src/main/java/co/cask/wrangler/service/connections/ConnectionService.java

```diff
@@ -234,12 +237,33 @@ public void list(HttpServiceRequest request, HttpServiceResponder responder,
         }
         return false;
       });
-      responder.sendJson(new ConnectionResponse<>(connections, connectionTypeConfig.getDefaultConnection()));
+      Optional<Connection> defaultConnection = getDefaultConnection(connections, connectionTypeConfig);
+      responder.sendJson(
+        new ConnectionResponse<>(connections,
+                                 defaultConnection.isPresent() ? defaultConnection.get().getName() : null));
     } catch (Exception e) {
       error(responder, e.getMessage());
     }
   }
 
+  private boolean isDefaultConnection(String connectionId) {
+    List<Connection> defaultConnections = connectionTypeConfig.getConnections();
+    // default connection provided as part of config only contains name and type of connection
+    // since connection id is derived from connection name, we get the connection-id for the default connection name
+    // and check with the input connection id.
+    return defaultConnections.stream()
+      .filter(c -> connectionId.equals(store.getConnectionId(c.getName()))).findFirst().isPresent();
+  }
+
+  private Optional<Connection> getDefaultConnection(List<Connection> connections,
```

Comment:
in this case, I think it's easier to just return null instead of using an Optional, as you end up converting to null anyway.